examples/.experimental/next-app-dir/src/app/standalone/_provider.tsx

```diff
@@ -0,0 +1,47 @@
+'use client';
+
+import type { inferTRPCClientOptionsDecoration } from '@trpc/client';
+import {
+  createTRPCClientOptions,
+  httpBatchLink,
+  loggerLink,
+} from '@trpc/client';
+import type { AppRouter } from '~/server/routers/_app';
+import { getUrl } from '~/trpc/shared';
+import superjson from 'superjson';
+import { cacheLink, refetchLink } from './_lib/cacheLink';
+import { createReactClient } from './_lib/createReactClient';
+
+const getTrpcOptions = createTRPCClientOptions<AppRouter>()(() => ({
+  links: [
+    loggerLink({
+      enabled: (op) => true,
+    }),
+    cacheLink(),
+    // FIXME: this is not working
+    // testDecorationLink(),
+    refetchLink(),
```

Comment:
Help appreciated here

basically, adding different links should change the options with the links' configuration

that will allow our client to be smart based on the functionality you add into links

links will be able to impact the options you pass to `.query`, e.g.

```ts
trpc.post.byId({
  refetchOnFocus: true,
})

if you add a `refetchOnFocusLink` that adds something

also, the `runtime`-object could become dynamic so that we could have the `cacheLink` append e.g. a `.cache` object to the runtime on initialization and allowing that object to be globally accessible when doing (de)hydration